### ISSUE TITLE
Investigate requirements to corpus pre-processing and recommended params to train fastText models for slavic languages

### ISSUE THREAD

**Issue by dchaplinsky** [lang-uk/word-vectors-uk#2]
Given the corpus size (665419885 tokens in ubertext 1.0 and roughly 3bln of tokens for ubertext 2.0) please prepare a list of recommendations on:
- texts pre-processing (digits, dates, tokens for the end of sentence/paragraph/article maybe)
- sets of params for training (there might be more than one set)

Describe it somewhere in the document in that repo (README.md is a good candidate), that's going to be our methodology :)

**Comment by MariRazno** [comment]
FastText text pre-processing:

- utf-8 encoding
- only alphabetic characters (clear digits, dates, punctuation)
- special characters for word boundaries
- new line character to delimit sentences

FastText data pre-processing:
https://fasttext.cc/docs/en/python-module.html#important-preprocessing-data--encoding-conventions

**Comment by MariRazno** [comment]
Parameters for FastText:
FastText's autotune feature allows to find automatically the best hyperparameters for each dataset.
In order to activate hyperparameter optimization, we must provide a validation file with the -autotune-validation argument.
Then, fastText will search the hyperparameters that gives the best f1-score.
https://fasttext.cc/docs/en/autotune.html#how-to-use-it

**Comment by dchaplinsky** [comment]
Digits/dates must be removed or replaced with special token? Does that rules apply to other algorithms as well? What is the special character for word boundaries?

**Comment by MariRazno** [comment]
Parameters of pre-trained FastText vectors on Ukrainian Wikipedia (https://aclanthology.org/L18-1550.pdf):
(wordNgrams=5-5, minn=2, maxn=5, dim=300, neg=10, epoch=10)

**Comment by dchaplinsky** [comment]
> Parameters for FastText:
> FastText's autotune feature allows to find automatically the best hyperparameters for each dataset.
> In order to activate hyperparameter optimization, we must provide a validation file with the -autotune-validation argument.
> Then, fastText will search the hyperparameters that gives the best f1-score.
> https://fasttext.cc/docs/en/autotune.html#how-to-use-it

How to collect validation file? Will the data that we have for intrinsic evaluation work?

**Comment by MariRazno** [comment]
> Will

I think it might work.

**Comment by MariRazno** [comment]
> > Parameters for FastText:
> > FastText's autotune feature allows to find automatically the best hyperparameters for each dataset.
> > In order to activate hyperparameter optimization, we must provide a validation file with the -autotune-validation argument.
> > Then, fastText will search the hyperparameters that gives the best f1-score.
> > https://fasttext.cc/docs/en/autotune.html#how-to-use-it
> 
> How to collect validation file? Will the data that we have for intrinsic evaluation work?

I think it might work.

**Comment by MariRazno** [comment]
> apply

**Comment by MariRazno** [comment]
> Digits/dates must be removed or replaced with special token? Does that rules apply to other algorithms as well? What is the special character for word boundaries?
> […](#)
> 11 лип. 2021 р. о 14:19 Maria ***@***.***> пише: ﻿ Text pre-processing: utf-8 encoding only alphabetic characters (clear digits, dates, punctuation) special characters for word boundaries new line character to delimit sentences FastText data pre-processing: https://fasttext.cc/docs/en/python-module.html#important-preprocessing-data--encoding-conventions — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

- Digits/Dates removal isn't a must but can positively influence on result. I think we can try two options (with digits/dates and without them). 

- FastText suggests:
  We advice the user to convert UTF-8 whitespace / word boundaries into one of the following symbols as appropiate.
    space
    tab
    vertical tab
    carriage return
    formfeed
    the null character
